src/lib/Imp.hs

```diff
@@ -73,22 +75,27 @@ data ImpCatEnv = ImpCatEnv
 type ImpM = ExceptT () (ReaderT ImpCtx (Cat ImpCatEnv))
 type AtomRecon = Abs (Nest Binder) Atom
 
-toImpModule :: Backend -> CallingConvention -> Name
+toImpModule :: TopEnv -> Backend -> CallingConvention -> Name
             -> [IBinder] -> Maybe Dest -> Block
             -> (ImpFunction, ImpModule, AtomRecon)
-toImpModule backend cc entryName argBinders maybeDest block =
-  runImpM initOpts inVarScope $ do
+toImpModule env backend cc entryName argBinders maybeDest block = do
+  let standaloneFunctions =
+        for (envPairs $ envIntersect (freeVars block) env) $
+           \(v, (_, LetBound _ (Atom f))) ->
+              runImpM initCtx inVarScope $ toImpStandalone v f
```

Comment:
Why do we use `inVarScope` here? This is the scoped used to compile the outer function, so why would it matter?

Reply:
It's a superset of the scope used to compile the standalone function. So it's still needed for fresh names etc.